src/runtime/types/memory.rs

```diff
@@ -16,6 +18,8 @@ pub const DEMI_SGARRAY_MAXLEN: usize = 1;
 #[repr(C, packed)]
 #[derive(Copy, Clone)]
 pub struct demi_sgaseg_t {
+    /// Reserved for DemiBuffer metadata.
+    pub sgaseg_md: *mut libc::c_void,
```

Comment:
We should evaluate renaming these fields to avoid redundant prefix `sgaseg`. That will also gives us more space to write meaningful names.

Reply:
Possibly... if we remove this prefix, we will have more space to write better/meaningful names for the fields (e.g., `reserved_metadata` would much better than `sgaseg_md`).